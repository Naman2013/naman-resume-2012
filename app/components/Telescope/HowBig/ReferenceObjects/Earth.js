import React from 'react';
import PropTypes from 'prop-types';
import earthImage from './svg/earth.svg';

const Earth = ({ x, y }) => (
  <g>
    <image xlinkHref={earthImage} width="100%" height="100%" />
  </g>
);


Earth.propTypes = {
  x: PropTypes.number,
  y: PropTypes.number,
  r: PropTypes.number,
};
Earth.defaultProps = {
  x: 0,
  y: 0,
  r: 50,
};
export default Earth;
