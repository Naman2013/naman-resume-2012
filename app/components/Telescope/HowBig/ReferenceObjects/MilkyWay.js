import React from 'react';

const MilkyWay = () => ();

export default MilkyWay;
