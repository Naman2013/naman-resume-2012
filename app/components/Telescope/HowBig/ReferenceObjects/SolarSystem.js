import React from 'react';

const SolarSystem = () => (
  <g></g>
);

export default SolarSystem;
