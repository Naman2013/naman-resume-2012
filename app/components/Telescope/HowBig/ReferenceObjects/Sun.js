import React from 'react';

const Sun = () => (
  <g></g>
);

export default Sun;
