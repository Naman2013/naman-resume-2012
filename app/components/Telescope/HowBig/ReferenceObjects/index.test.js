import React from 'react';
import { shallow } from 'enzyme';
import Earth from './Earth';
import MilkyWay from './MilkyWay';
import SolarSystem from './SolarSystem';
import Sun from './Sun';

describe('Earth', () => {
  const earth = shallow(<Earth x={20} y={20} />);
  it('should render correctly', () => {
    expect(earth).toMatchSnapshot();
  });
});

describe('MilkyWay', () => {
  const milkyWay = shallow(<MilkyWay />);
  it('should render correctly', () => {
    expect(milkyWay).toMatchSnapshot();
  });
});

describe('SolarSystem', () => {
  const solarSystem = shallow(<SolarSystem />);
  it('should render correctly', () => {
    expect(solarSystem).toMatchSnapshot();
  });
});

describe('Sun', () => {
  const sun = shallow(<Sun />);
  it('should render correctly', () => {
    expect(sun).toMatchSnapshot();
  });
});
