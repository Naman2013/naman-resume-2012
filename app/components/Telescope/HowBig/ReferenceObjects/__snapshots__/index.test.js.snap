// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Earth should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Earth
    r={100}
    x={20}
    y={20}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <circle
        cx={20}
        cy={20}
        r={100}
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "cx": 20,
        "cy": 20,
        "r": 100,
      },
      "ref": null,
      "rendered": null,
      "type": "circle",
    },
    "type": "g",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <circle
          cx={20}
          cy={20}
          r={100}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "cx": 20,
          "cy": 20,
          "r": 100,
        },
        "ref": null,
        "rendered": null,
        "type": "circle",
      },
      "type": "g",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`MilkyWay should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MilkyWay />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": "g",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": "g",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SolarSystem should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SolarSystem />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": "g",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": "g",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Sun should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Sun />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": "g",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": "g",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
